### ISSUE TITLE
Linux build stopped

### ISSUE THREAD

**Issue by stwe** [Hatrickek/OxylusEngine#9]
I'm getting some errors when compiling. Here is a selection:

``` bash
error: declaration of ‘void operator delete(void*, std::size_t) noexcept’ has a different exception specifier
  135 | void operator delete(void*, std::size_t) _GLIBCXX_USE_NOEXCEPT
      |      ^~~~~~~~
```

```bash
error: declaration of ‘Oxylus::Ref<Oxylus::Mesh> Oxylus::ShapeInfo::Mesh’ changes meaning of ‘Mesh’ [-fpermissive]
   28 |     Ref<Mesh> Mesh = nullptr;
      |               ^~~~
```

```bash
error: declaration of ‘Oxylus::Ref<Oxylus::RenderGraph> Oxylus::VulkanRenderer::RendererContext::RenderGraph’ changes meaning of ‘RenderGraph’ [-fpermissive]
   29 |       Ref<RenderGraph> RenderGraph = nullptr;
      |                        ^~~~~~~~~~~
```

Maybe you can put a tag on a working version.

In addition, you 3rd Party Libs is firmly integrated into your project. Maybe it would be better to get them with VCPKG or some other package manager.

**Comment by Hatrickek** [comment]
Looks like you are using GCC which is not currently supported or had ever been tested. On linux only Clang is supported.

Currently we use CMake fetchcontent and submodules to manage our dependencies which is a good enough way to manage them. In the future I'm thinking of ditching all submodules and just using fetchcontent for all dependencies.

**Comment by Hatrickek** [comment]
fixed with 750bc04
